### DISCUSSION TITLE
How do I use this with numtide/nix-filter?

### DISCUSSION THREAD

**Discussion by nothingnesses** [90-008/nix-cargo-integration#124] [Q&A] [answered]
I'm trying to update [the flake.nix I wrote for this project](https://github.com/nothingnesses/LambdaVM-Rust/blob/4ac0acb5d02e7b6f2f56e8d0b76aaea89b34b92c/flake.nix) which is currently using an older version of NCI not based on flake-parts. So far I've come up with [this flake.nix](https://gist.github.com/nothingnesses/113283b36f36ad863fa92bfd5f598849/c04ac48f48cce6c0a3abae030b354074324cc029). Unfortunately, this new version fails to evaluate (e.g., when trying to access the dev shell with `nix develop --impure`) and instead results in [the same error as in NCI's readme](https://github.com/yusdacra/nix-cargo-integration/tree/c17e8014b379035dbbb4b244a58fac4b39aed756#ignoring-cargolock-in-rust-libraries), I think because the outputs of nix-filter doesn't use git.

I can get the flake to evaluate if I replace [these lines](https://gist.github.com/nothingnesses/113283b36f36ad863fa92bfd5f598849#file-flake-nix-L42-L45) with `nci.projects.${crate_name}.relPath = "";` instead, but obviously this would cause the derivation to include a bunch of other unnecessary files. How could I get this working with [numtide/nix-filter](https://github.com/numtide/nix-filter)?

**Comment by 90-008** [comment, marked as answer]
This wasn't possible before (because I forgot to add an option to actually set the source) but you can now do this by setting nci.source to your nix-filtered source and leave relPath as "". If you update to latest master you should be able to use it.

**Comment by nothingnesses** [comment]
Thank you for the quick feature implementation, [using nci.source](https://gist.github.com/nothingnesses/113283b36f36ad863fa92bfd5f598849/3e7112ba1916e8dfa17b9b951f5631feb0dc575b) seems to work nicely!
